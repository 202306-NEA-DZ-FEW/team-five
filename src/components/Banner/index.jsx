const Banner = ({ description, title, title1, imageUrl, btnText, title2 }) => {
    const imgStyle = {
        backgroundImage: "url(" + imageUrl + ")",
    };
    return (
        <section>
            <div
                className='bg-cover bg-center h-[30vh] sm:h-[50vh] md:h-[70vh] text-white  object-fill'
                style={imgStyle}
            >
                <div className='bg-opacity-50 bg-black h-full items-center flex justify-center flex-col'>
                    <div className='text-center'>
                        <h1 className='text-xl sm:text-3xl md:text-5xl font-bold text-green-500'>
                            {title}
                            <span className='text-white'>{title1}</span>
                            <br />
                            {title2}
                        </h1>
                        <p className='mt-6 sm:mt-8 md:mt-10 text-center text-[9px] sm:text-xs md:text-xl '>
                            {description}
                        </p>
                    </div>
                    <div className='mt-6 sm:mt-8 md:mt-10'>
                        <button className='bg-green-500 rounded-full p-2 px-3 text-[9px] sm:text-base sm:px-6 text-center cursor-pointer font-medium hover:bg-green-700'>
                            {btnText}
                        </button>
                    </div>
                </div>
            </div>
        </section>
    );
};

export default Banner;
