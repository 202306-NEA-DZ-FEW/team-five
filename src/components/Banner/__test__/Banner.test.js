import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";

import Banner from "..";

jest.mock("next/router", () => ({
    useRouter: () => ({
        route: "/",
        pathname: "",
        query: {},
        asPath: "",
    }),
}));

jest.mock("react-i18next", () => ({
    useTranslation: () => ({ t: (key) => key, i18n: { language: "en" } }),
}));

const mockStore = configureMockStore([]);
const store = mockStore({
    userInfo: null,
});

describe("Banner Component", () => {
    it("should render correctly", () => {
        const component = renderer.create(
            <Provider store={store}>
                <Banner />
            </Provider>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
