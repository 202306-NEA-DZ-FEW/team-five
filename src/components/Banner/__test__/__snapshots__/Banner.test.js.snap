// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Component should render correctly 1`] = `
<section>
  <div
    className="bg-cover bg-center h-[30vh] sm:h-[50vh] md:h-[70vh] text-white  object-fill"
    style={
      Object {
        "backgroundImage": "url(undefined)",
      }
    }
  >
    <div
      className="bg-opacity-50 bg-black h-full items-center flex justify-center flex-col"
    >
      <div
        className="text-center"
      >
        <h1
          className="text-xl sm:text-3xl md:text-5xl font-bold text-[#20DC49]"
        >
          <span
            className="text-white"
          />
          <br />
        </h1>
        <p
          className="mt-6 sm:mt-8 md:mt-10 text-center text-[9px] sm:text-xs md:text-xl "
        />
      </div>
      <div
        className="mt-6 sm:mt-8 md:mt-10"
      >
        <a
          href="/coupons"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <button
            className="bg-[#20DC49] rounded-full p-2 px-6 text-[9px] sm:text-base sm:px-6 text-start cursor-pointer font-medium hover:bg-green-600"
          />
        </a>
      </div>
    </div>
  </div>
</section>
`;
